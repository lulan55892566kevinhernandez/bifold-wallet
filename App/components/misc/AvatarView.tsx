import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import { hashCode, hashToRGBA } from '../../utils/helpers'
import { testIdWithKey } from '../../utils/testable'
import { useThemeContext } from '../../utils/themeContext'

interface AvatarViewProps {
  name: string
}

const AvatarView: React.FC<AvatarViewProps> = ({ name }) => {
  const { TextTheme } = useThemeContext()
  const styles = StyleSheet.create({
    avatar: {
      width: TextTheme.headingTwo.fontSize * 2,
      height: TextTheme.headingTwo.fontSize * 2,
      margin: 12,
      borderWidth: 3,
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: TextTheme.headingTwo.fontSize,
      borderColor: TextTheme.headingTwo.color,
    },
  })
  return (
    <View style={[styles.avatar, { borderColor: hashToRGBA(hashCode(name)) }]}>
      <Text style={[TextTheme.headingTwo, { fontWeight: 'normal' }]} testID={testIdWithKey('AvatarName')}>
        {name.charAt(0)}
      </Text>
    </View>
  )
}

export default AvatarView
