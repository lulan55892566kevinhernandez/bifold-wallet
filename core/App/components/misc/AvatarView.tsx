import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import { hashCode, hashToRGBA } from '../../utils/helpers'
import { testIdWithKey } from '../../utils/testable'
import { useThemeContext } from '../../utils/themeContext'

interface AvatarViewProps {
  name: string
}

const AvatarView: React.FC<AvatarViewProps> = ({ name }) => {
  const { TextTheme, ListItems } = useThemeContext()
  const styles = StyleSheet.create({
    avatar: {
      ...ListItems.avatarCircle,
      margin: 12,
      borderWidth: 3,
      justifyContent: 'center',
      alignItems: 'center',
    },
  })
  return (
    <View style={[styles.avatar, { borderColor: hashToRGBA(hashCode(name)) }]}>
      <Text style={ListItems.avatarText} testID={testIdWithKey('AvatarName')}>
        {name.charAt(0)}
      </Text>
    </View>
  )
}

export default AvatarView
