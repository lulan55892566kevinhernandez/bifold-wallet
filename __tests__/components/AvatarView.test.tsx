import { render } from '@testing-library/react-native'
import React from 'react'

import AvatarView from '../../App/components/misc/AvatarView'

describe('AvatarView Component', () => {
  test('Renders correctly', async () => {
    const tree = render(<AvatarView name={'Bacon'} />)

    expect(tree).toMatchSnapshot()
  })
})
