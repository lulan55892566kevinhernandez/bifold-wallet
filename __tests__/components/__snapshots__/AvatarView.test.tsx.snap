// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarView Component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 32,
        "borderWidth": 3,
        "height": 64,
        "justifyContent": "center",
        "margin": 12,
        "width": 64,
      },
      Object {
        "borderColor": "#CFA8C3",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 32,
          "fontWeight": "normal",
        },
      ]
    }
  >
    B
  </Text>
</View>
`;
